src/shared/extensionUtilities.ts

```diff
@@ -109,6 +132,23 @@ export function isCloud9(flavor: 'classic' | 'codecatalyst' | 'any' = 'any'): bo
     return (flavor === 'classic' && !codecat) || (flavor === 'codecatalyst' && codecat)
 }
 
+export function isSageMaker(): boolean {
+    return vscode.env.appName === sageMakerAppname
+}
+
+/**
+ * Returns true if credentials are provided by the environment (ex. via ~/.aws/)
+ *
+ * @param isC9 boolean for if Cloud9 is host
+ * @param isSM boolean for if SageMaker is host
+ * @returns boolean for if C9 "OR" SM
+ */
+export function hasVendedIamCredentials(isC9?: boolean, isSM?: boolean) {
```

Comment:
would you mind moving this to [auth.ts](https://github.com/aws/aws-toolkit-vscode/blob/13ebf3aee88b2357a9a59b27d0308417b3f105c0/src/auth/auth.ts)